www/caddy/src/opnsense/mvc/app/controllers/Pischem/Caddy/Api/ReverseProxyController.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+/**
+ *    Copyright (C) 2023-2024 Cedrik Pischem
+ *    Copyright (C) 2015 Deciso B.V.
+ *
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *
+ *    1. Redistributions of source code must retain the above copyright notice,
+ *       this list of conditions and the following disclaimer.
+ *
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *
+ *    THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *    INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ *    AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *    AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ *    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+namespace Pischem\Caddy\Api;
+
+use OPNsense\Base\ApiMutableModelControllerBase;
+
+class ReverseProxyController extends ApiMutableModelControllerBase
+{
+    protected static $internalModelName = 'caddy';
+    protected static $internalModelClass = 'Pischem\Caddy\Caddy';
+
```

Comment:
Hi )
Sorry if I missed something again, are you sure you don't want to enable  $internalModelUseSafeDelete?
https://github.com/opnsense/core/blob/875fb5d8b3be347265bc86a6115c2d653c2fa1a6/src/opnsense/mvc/app/controllers/OPNsense/Base/ApiMutableModelControllerBase.php#L93

Reply:
Hmm, this sounds interesting. I put a lot of logic into the template though that generates the Caddyfile. Users can delete anything they want and it won't results in a broken config. So I think I can leave this disabled? What's your opinion on that?

Reply:
hm. this of course depends on the logic you put. you can make an error-proof template but what about config references? Maybe @fichtner  will correct me, but I thought that is correct to either provide a cascade deleting on the controller del action or not allow deleting while there are refernces to the object - otherwise there will be orphan references in config?

Reply:
Im using a lot of model relation fields. If Caddy configs are very large it might be a good help to have this enabled.

Reply:
I think the error message is trying to show the value of the `description` (if exists) of the object using the element being removed.
I haven't checked the error-tolerance of caddy template (will an empty dict error appear if there are links only to deleted elements?), I just think that you never know when orphan links will bite you in the ass )

Reply:
Since I prefer my ass not being bitten, I have added it: 700cb8f

Thank you for the feedback. :+1:

Reply:
@kulikov-a I think it only checks for exactly `description` in lower case. That's why no description is shown with the error. All of my keys in `Caddy.xml` are `Description` with an uppercase `D`. I once had the same happening with `enabled` and `Enabled` (its mandatory that its lower case since otherwise the enabled checkbox doesn't work), but I didn't check that description /had/ to be in lower case. I don't know how to fix this now so late in the process with many people having stored their data in "Description" already, and it being a required field due to the references of the model relation fields.

Reply:
Migration from Description to description: 18dc214

Update worked on my production Firewall with Caddy. Now all `$internalModelUseSafeDelete` show the correct description on error.

Reply:
@Monviech hi, sorry for missing (39° is not fun, as it turns out). Glad you found a way to solve this problem with a capitalized key.
I also don't know a nice way to do this check case-insensitive (casting an object to an array to use the `array_change_key_case` doesn't look very nice), so I'd suggest this to @AdSchellevis, sorry

Reply:
@kulikov-a That sounds hot indeed. Don't worry the problem is already solved with a Migration script I wrote that migrates all the "Description" key values to "description" key values. So this problem is solved. Thanks for checking up on this. :)